src/main/java/com/teragrep/pth_10/steps/teragrep/TeragrepDynatraceStep.java

```diff
@@ -159,35 +159,52 @@ private void sendPostReq(String urlStr, DynatraceItem dti) throws IOException {
 
-            try (InputStream respStream = response.getEntity().getContent()) {
-                JsonObject jsonResp = new Gson()
-                        .fromJson(new InputStreamReader(respStream, StandardCharsets.UTF_8), JsonObject.class);
-                JsonElement errorElem = jsonResp.get("error");
-                if (!(errorElem instanceof JsonNull)) {
-                    throw new RuntimeException("Error from server response: " + errorElem.toString());
-                }
-                JsonElement validElem = jsonResp.get("linesValid");
-                if (validElem == null) {
-                    throw new RuntimeException("Unexpected JSON: Could not find linesValid element.");
+            final HttpEntity entity = response.getEntity();
+
+            if (entity == null) {
+                throw new IllegalStateException("Response entity was null!");
+            }
+
+            final JsonObject jsonResp;
+            try (final InputStream respStream = entity.getContent()) {
+                final InputStreamReader inputStreamReader = new InputStreamReader(respStream, StandardCharsets.UTF_8);
+                try {
+                    jsonResp = new Gson().fromJson(inputStreamReader, JsonObject.class);
                 }
-                LOGGER.info("Valid lines: <[{}]>", validElem);
-                JsonElement invalidElem = jsonResp.get("linesInvalid");
-                if (invalidElem == null) {
-                    throw new RuntimeException("Unexpected JSON: Could not find linesInvalid element.");
+                catch (final JsonIOException | JsonSyntaxException je) {
+                    throw new IllegalArgumentException("Error parsing JSON response, message: " + je.getMessage());
                 }
-                LOGGER.warn("Invalid lines: <[{}]>", invalidElem);
             }
+            catch (final IOException | UnsupportedOperationException e) {
+                throw new IllegalStateException("Error getting response stream, message: " + e.getMessage());
+            }
+            if (jsonResp == null || jsonResp.isJsonNull()) {
+                throw new IllegalStateException("Unexpected null JSON response");
+            }
+            JsonElement errorElem = jsonResp.get("error");
+            if (!errorElem.isJsonNull()) {
+                throw new RuntimeException("Error from server response: " + errorElem.toString());
+            }
+            JsonElement validElem = jsonResp.get("linesValid");
+            if (validElem == null) {
+                throw new RuntimeException("Unexpected JSON: Could not find linesValid element.");
+            }
+            LOGGER.info("Valid lines: <[{}]>", validElem);
+            JsonElement invalidElem = jsonResp.get("linesInvalid");
+            if (invalidElem == null) {
+                throw new RuntimeException("Unexpected JSON: Could not find linesInvalid element.");
+            }
+            LOGGER.warn("Invalid lines: <[{}]>", invalidElem);
```

Comment:
```suggestion
            
            if (jsonResp.has("error") && !jsonResp.get("error").isJsonNull()) {
                throw new RuntimeException("Error from server response: " + jsonResp.get("error").toString());
            }
            
            final boolean hasValidElem = jsonResp.has("linesValid");
            if (!hasValidElem) {
                throw new RuntimeException("Unexpected JSON: Could not find linesValid element.");
            }
            LOGGER.info("Valid lines: <[{}]>", jsonResp.get("linesValid"));
            
            final boolean hasInvalidElem = jsonResp.has("linesInvalid");
            if (!hasInvalidElem) {
                throw new RuntimeException("Unexpected JSON: Could not find linesInvalid element.");
            }
            LOGGER.warn("Invalid lines: <[{}]>", jsonResp.get("linesInvalid"));
```

This way null checks are not necessary